src/app/state/middleware/notificationManager.ts

```diff
@@ -29,6 +30,11 @@ export const notificationCheckerMiddleware: Middleware = (middlewareApi: Middlew
 
         if (isDefined(user)) {
             const firstLogin = persistence.session.load(KEY.FIRST_LOGIN)
+            // email confirmation modal to take precedence over other modals, only for teacherPending accounts
+            if (needToVerifyEmail(user.teacherPending, user.emailVerificationStatus) && !withinLast2Minutes(persistence.load(KEY.EMAIL_CONFIRMATION_MODAL_SHOWN_TIME)) && firstLogin === null) {
```

Comment:
did we check the nagging behaviour with STEM? Just wondering if there's any delay on the manual verification step at STEM, the 2 minute nag could be really painful.

Reply:
This pop-up is for the email verification, not the teacher pending review. So the user is able to confirm their email address immediately and they will still have to wait up to 5 days for the teacher pending flag to be reviewed during which the pop-up will not keep annoying them